docs/docs/guides/03-typescript.md

```diff
@@ -2,4 +2,266 @@
+
+## 2. Modules, Imports and Exports
+
+### a. Modules
+
+In TypeScript, modules can be used to organize and encapsulate related types and code. Modules provide a way to mimic namespaces and avoid polluting the global scope with numerous type declarations. By grouping related types within modules, it is possible to improve code organization and make it easier to navigate and understand the codebase.
+
+This is especially true when working with packages exports or utilities, where the exported components needs to be scoped.
+
+Example:
+
+```typescript
+// service.d.ts
+export type Admin = Record<string, any>;
+
+// uid.d.ts
+export type Admin = `admin::${string}`;
+
+// index.d.ts
+export * as Service from './service';
+export * as UID from './uid';
+
+// src/index.ts
+import type { Service, UID } from './index';
+
+type AdminService = Record<UID.Admin, Service.Admin>;
+```
+
+### b. Imports
+
+When importing **only types** from a module, `import type` must be used.
+
+The `import type` syntax imports only the types and eliminates the runtime impact of importing the actual values. This helps reduce the bundle size and improves performance by avoiding unnecessary code execution.
+
+Example:
+
+```typescript
+// ❌ don't
+
+import { Attribute, Common } from '@strapi/strapi';
+
+// ✅ do
+
+import type { Attribute, Common } from '@strapi/strapi';
+```
+
+When importing **both types and implementations** from a module:
+
+- `import type` must be used for types
+- `import` must be used for the rest
+
+Example:
+
+```typescript
+// ❌ don't
+
+import { User, findUser } from './user';
+
+const user: User = findUser();
+
+// ✅ do
+
+import type { User } from './user';
+import { findUser } from './user';
+
+const user: User = findUser();
+```
+
+### c. Exports
+
+In most cases, it is recommended to use named exports (`export`) instead of the default export (`export default`). Named exports provide more flexibility and clarity when importing and using modules. They allow importing specific members of a module explicitly, which improves code readability and maintainability.
+
+However, there are a few exceptions where using export default might be appropriate. For example, when exporting a single primary entity or class from a module, or when working with modules that have a natural default export, such as a configuration object.
+
+Using regular exports provides a clearer and more explicit interface for module consumers, promoting better code organization and maintainability. However, it is important to consider the context before deciding whether to use default exports.
+
+## 3. Parameters for Generics
+
+When working with generic types, it is required to follow naming conventions and use descriptive names for type parameters. This helps improve code readability and maintainability over the whole codebase.
+
+- For simple types with a single self-explanatory parameter, it is allowed to **use `T`**
```

Comment:
Could we say "For generic type declaration with a single parameter" 🤔  so we make it clearer we only allow it for pure ts declaration and not inside code.

Same question as above what about eslint ?

Reply:
> and not inside code.

What about

```ts
function doSomething<T extends string>(value: T): T {
  // ...
}
```

It's code, but I don't think it makes sense to name the generic parameter